core/src/main/java/org/apache/brooklyn/core/network/AbstractOnNetworkEnricher.java

```diff
@@ -0,0 +1,402 @@
+            Optional<HostAndPort> mappedEndpoint = getMappedEndpoint(source, machine, port);
+            if (!mappedEndpoint.isPresent()) {
+                LOG.debug("network-facing enricher not transforming {} host-and-port {}, because no port-mapping for {}", new Object[] {source, sensorVal, machine});
+                return Maybe.absent();
+            }
+            if (!mappedEndpoint.get().hasPort()) {
+                LOG.debug("network-facing enricher not transforming {} host-and-port {}, because no port in target {} for {}", new Object[] {source, sensorVal, mappedEndpoint, machine});
+                return Maybe.absent();
+            }
+            return Maybe.of(mappedEndpoint.get().toString());
+        } else {
+            LOG.debug("network-facing enricher not transforming {} host-and-port {} because defines no port", source, hostAndPort);
+            return Maybe.absent();
+        }
+    }
+
+    protected Maybe<String> transformPort(Entity source, MachineLocation machine, int sensorVal) {
+        if (Networking.isPortValid(sensorVal)) {
+            Optional<HostAndPort> mappedEndpoint = getMappedEndpoint(source, machine, sensorVal);
+            if (!mappedEndpoint.isPresent()) {
+                LOG.debug("network-facing enricher not transforming {} port {}, because no port-mapping for {}", new Object[] {source, sensorVal, machine});
+                return Maybe.absent();
+            }
+            if (!mappedEndpoint.get().hasPort()) {
+                LOG.debug("network-facing enricher not transforming {} port {}, because no port in target {} for {}", new Object[] {source, sensorVal, mappedEndpoint, machine});
+                return Maybe.absent();
+            }
+            return Maybe.of(mappedEndpoint.get().toString());
+        } else {
+            LOG.debug("network-facing enricher not transforming {} port {} because not a valid port", source, sensorVal);
+            return Maybe.absent();
+        }
+    }
+
+    protected Maybe<MachineLocation> getMachine() {
+        return Machines.findUniqueMachineLocation(entity.getLocations());
+    }
+    
+    protected void checkConfig() {
+        Collection<? extends AttributeSensor<?>> sensors = getConfig(SENSORS);
+        Maybe<Object> rawMapMatching = config().getRaw(MAP_MATCHING);
+        String mapMatching = config().get(MAP_MATCHING);
+        
+        if (sensors == null || sensors.isEmpty()) {
+            if (Strings.isBlank(mapMatching)) {
+                throw new IllegalStateException(this+" requires 'sensors' config (when 'mapMatching' is explicitly blank)");
+            }
+        } else if (rawMapMatching.isPresent()) {
+            throw new IllegalStateException(this+" must not have explicit 'mapMatching' and 'sensors' config");
+        }
+    }
+    
+    protected Collection<AttributeSensor<?>> resolveSensorsConfig() {
+        Collection<? extends AttributeSensor<?>> sensors = getConfig(SENSORS);
```

Comment:
Could've gone one step further and accept both `String` and `List` after removing `SENSOR`.